elastic/kibana
# [Index Management] Disable certain actions for serverless

Fixes https://github.com/elastic/kibana/issues/157945

### Summary

In serverless, certain index actions will be blocked by Elasticsearch. This PR creates a new config that allows us to configure their visibility to the user and disables it for serverless. The disabled actions are:
- /{index}/_cache/clear
- /{index}/_flush
- /{index}/_forcemerge
- /{index}/_freeze
- /{index}/_unfreeze
- /{index}/_stats
- /{index}/_close (internal access only)
- /{index}/_open (internal access only)
- /{index}/_refresh


##### How to test
1. Start elasticsearch with `yarn es snapshot --license=trial` and serverless search solution with `yarn serverless-es`
2. Navigate to `Console` and create a simple index `PUT test/_doc/1 { message: "hello" }`
3. Navigate to the `Index Management` app (Navigation doesnt include a link to it yet but can can access it by url directly with `/app/management/data/index_management/indices`) and make sure that the actions are limited to whats allowed.
4. Check in regular on-prem kibana that all the actions are available


<details>
<summary>Screenshots</summary>

// Serverless
![Screenshot 2023-07-10 at 10 36 53](https://github.com/elastic/kibana/assets/1191206/38b2373d-afb1-4d55-a889-3ac8146b6278)
// On-prem
![Screenshot 2023-07-10 at 10 32 55](https://github.com/elastic/kibana/assets/1191206/18464def-9fef-402e-8aed-89f4d17cac2b)

</details>
